### DISCUSSION TITLE
till what degree Pocketbase is stable as this date to be used in projects

### DISCUSSION THREAD

**Discussion by anasfik** [pocketbase/pocketbase#4074] [Q&A]
Hello,

Thank you for your amazing work.

I have been following Pocketbase for a while, always wanting to self host and use it with Flutter, I know it is on heavy development right now so my question is to what degree it is usable for production for the auth, database, and file uploads...

can I use it in projects and expect it to be working fine for what it has been implemented so far?

**Comment by ganigeorgiev** [comment]
It is stable in the sense that there shouldn't be any major bugs.
If such bugs are found they are fixed as soon as they are identified (_note that bug fixes are not back ported and they are applied only for new versions_).

It is not stable in the sense that backward compatibility is not guaranteed and you may need to apply some manual migration steps before updating (usually prefixed with ⚠️ in the [CHANGELOG](https://github.com/pocketbase/pocketbase/blob/master/CHANGELOG.md)).
I'm not planning major changes to the web APIs (_with exception maybe the auth flow and the (T)OTP handling_) but there will be breaking changes to the exposed Go APIs and hooks to make it a better framework. I'll try to provide more information on this soon after the release of Presentator, my other open source project (see https://github.com/pocketbase/pocketbase/discussions/3271).

**Comment by jonshipman** [comment]
As with any software, if you lock it to the current version it will act exactly as it is right now until the end of time. That's not to say that some vulnerability or exploit won't be found in a month or two.

IME, sqlite is best suited for simple single-purpose apps. In that case, you should be fine. If you're building a Facebook clone you might want to look elsewhere.

Thus far, I've used Pocketbase for at least 5 client projects and dozens of personal experiments.
